applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -60,368 +57,300 @@ public ConfigurationClient(String connectionString, HttpPipeline pipeline) {
-            throw new IllegalArgumentException("Parameter key is required and cannot be null or empty");
+            throw new IllegalArgumentException("Parameter 'key' is required and cannot be null or empty");
+        } else if (label == null) {
+            label = ConfigurationSetting.NULL_LABEL;
         }
-        if (filter != null) {
-            return service.unlockKeyValue(baseUri.toString(), key, label, filter.ifMatch(), filter.ifNoneMatch());
-        }
-        return service.unlockKeyValue(baseUri.toString(), key, label, null, null);
-    }
 
-    /**
-     * Lists chronological/historical representation of ConfigurationSetting resource(s). Revisions eventually expire (default 30 days).
-     * For all operations key is optional parameter. If ommited it implies any key.
-     * For all operations label is optional parameter. If ommited it implies any label.
-     * @param filter query options
-     * @return Revisions of the ConfigurationSetting
-     */
-    public Flux<ConfigurationSetting> listKeyValueRevisions(RevisionFilter filter) {
-        return listKeyValueRevisions(filter, pageResponseFlux -> pageResponseFlux.map(r -> r.items())
-                                                              .flatMapIterable(i -> i));
-    }
-
-    public <T> Flux<T> listKeyValueRevisions(RevisionFilter filter, Function<Flux<RestPagedResponse<ConfigurationSetting>>, ? extends Flux<T>> receiver) {
-        Flux<RestPagedResponse<ConfigurationSetting>> p = listRevisionsSinglePageAsync(filter)
-                       .concatMap(page -> {
-                           String nextPageLink = page.nextLink();
-                           if (nextPageLink == null) {
-                               return Flux.just(page);
-                           }
-                           return Flux.just(page).concatWith(listNextAsync(nextPageLink));
-                       });
-        return receiver.apply(p);
+        return service.unlockKeyValue(baseUri.toString(), key, label, null, null);
     }
 
     /**
-     * Gets all Revisions for ConfigurationSetting(s).
+     * Lists the ConfigurationSettings.
      *
-     * @return the Single&lt;Page&lt;ConfigurationSetting&gt;&gt; object if successful.
+     * @param filter query options
+     * @return KeyValues
      */
-    private Flux<RestPagedResponse<ConfigurationSetting>> listRevisionsSinglePageAsync(RevisionFilter filter) {
-        Mono<RestResponse<PageImpl<ConfigurationSetting>>> result;
+    public Flux<ConfigurationSetting> listKeyValues(KeyValueListFilter filter) {
+        Mono<RestResponse<Page<ConfigurationSetting>>> result;
         if (filter != null) {
-            result = service.listKeyValueRevisions(baseUri.toString(), filter.key(), filter.label(), filter.fields(), filter.acceptDatetime(), filter.range());
+            result = service.listKeyValues(baseUri.toString(), filter.key(), filter.label(), filter.fields(), filter.acceptDateTime(), filter.range());
```

Comment:
I thought the service team agreed on only having `Range` for the Revisions API ... https://msazure.visualstudio.com/Azure%20AppConfig/_git/AppConfigService?path=%2Fdocs%2Fprotocol%2Frevisions.md&version=GBmaster

Does the range work for `listKeyValues`?